stack/global/cloudtrail.tf

```diff
@@ -1,20 +1,22 @@
 data "aws_caller_identity" "current" {}
 
 module "log-producer" {
+  count  = var.cloudtrail_config == null ? 0 : 1
   source = "../../cloudtrail/log-producer"
 
   environment                        = var.environment
   project                            = var.project
   is_organization_trail              = false
   is_multi_region_trail              = true
   enable_management_cloudtrail       = true
-  enable_data_cloudtrail             = var.enable_data_cloudtrail
-  s3_bucket_arns_for_data_cloudtrail = var.s3_bucket_arns_for_data_cloudtrail
-  cloudtrail_target_bucket_name      = module.log-ingestor.s3_bucket_name
-  cloudtrail_target_bucket_kms_arn   = module.log-ingestor.s3_kms_arn
+  enable_data_cloudtrail             = var.cloudtrail_config.enable_data_cloudtrail
+  s3_bucket_arns_for_data_cloudtrail = var.cloudtrail_config.s3_bucket_arns_for_data_cloudtrail
+  cloudtrail_target_bucket_name      = module.log-ingestor[0].s3_bucket_name
+  cloudtrail_target_bucket_kms_arn   = module.log-ingestor[0].s3_kms_arn
 }
 
 module "log-ingestor" {
+  count  = var.cloudtrail_config == null ? 0 : 1
```

Comment:
I think we should have a different boolean for the `log-ingestor`.

Example 1:
Assume we have 2 accounts, then we launch 2 `stack/global` instances. In both we will need `log-producer`, but only in 1 we will need `log-ingestor`. 

Example 2:
Assume all logs are collected in 1 account that only collects logs. In all other accounts, we launch 1 `stack/global` module, and the only thing we need for Cloutrail is to produce logs, but not to ingest.

Reply:
> In both we will need log-producer, but only in 1 we will need log-ingestor.

Assume we have 2 accounts, we will only need 
Account A: 1 log-producer (cloudtrail only)
Account B: 1 log-ingester (cloudtrail + s3)
Account C: 1 log-producer (cloudtrail only)

Right?

Reply:
Correct!